test-app/src/main/java/com/mapbox/navigation/examples/core/MapboxNavigationActivity.kt

```diff
@@ -0,0 +1,409 @@
+        )
+        init()
+        tripProgressApi = MapboxTripProgressApi(getTripProgressFormatter())
+        maneuverApi = MapboxManeuverApi(
+            MapboxDistanceFormatter(DistanceFormatterOptions.Builder(this).build())
+        )
+    }
+
+    override fun onStart() {
+        super.onStart()
+        mapView.onStart()
+        if (::mapboxNavigation.isInitialized) {
+            mapboxNavigation.registerRoutesObserver(routesObserver)
+        }
+    }
+
+    override fun onStop() {
+        super.onStop()
+        if (::mapboxNavigation.isInitialized) {
+            mapboxNavigation.unregisterRoutesObserver(routesObserver)
+        }
+        navigationCamera.resetFrame()
+        mapView.onStop()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        mapView.onDestroy()
+        mapboxNavigation.onDestroy()
+    }
+
+    override fun onLowMemory() {
+        super.onLowMemory()
+        mapView.onLowMemory()
+    }
+
+    override fun onMapLongClick(point: Point): Boolean {
+        val currentLocation = navigationLocationProvider.lastLocation
+        if (currentLocation != null) {
+            val originPoint = Point.fromLngLat(
+                currentLocation.longitude,
+                currentLocation.latitude
+            )
+            findRoute(originPoint, point)
+        }
+        return false
+    }
+
+    private fun init() {
+        initNavigation()
+        initStyle()
+        slackLine.initialize(mapView, mapboxNavigation)
+        initViews()
+    }
+
+    private fun initViews() {
+        start.setOnClickListener {
+            startNavigation()
+            start.visibility = GONE
+            maneuverView.visibility = VISIBLE
+            tripProgressCard.visibility = VISIBLE
+        }
+        stop.setOnClickListener {
+            stopNavigation()
+            maneuverView.visibility = GONE
+            tripProgressCard.visibility = GONE
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun initNavigation() {
+        mapboxNavigation = MapboxNavigation(
+            NavigationOptions.Builder(this)
+                .accessToken(getMapboxAccessTokenFromResources())
+                .locationEngine(ReplayLocationEngine(mapboxReplayer))
+                .build()
+        ).apply {
+            registerLocationObserver(object : LocationObserver {
+                override fun onRawLocationChanged(rawLocation: Location) {
+                    updateCameraToIdle()
+                    val point = Point.fromLngLat(rawLocation.longitude, rawLocation.latitude)
+                    val cameraOptions = CameraOptions.Builder()
+                        .center(point)
+                        .zoom(13.0)
+                        .pitch(45.0)
+                        .build()
+                    mapboxMap.easeTo(cameraOptions)
+                    navigationLocationProvider.changePosition(rawLocation)
+                    mapboxNavigation.unregisterLocationObserver(this)
```

Comment:
This isn't necessarily needed, we should just request the camera to following after the first map matcher result is available.

Reply:
I am not using a `map matcher` observer in this example and hence doing this to animate the camera to my current location in the beginning of the app.